app/scripts/modules/kubernetes/src/validation/applicationName.validator.js

```diff
@@ -10,11 +10,16 @@ module.exports = angular
   ])
   .factory('kubernetesApplicationNameValidator', function () {
 
-    function validateSpecialCharacters(name, errors) {
+    function validateSpecialCharacters(name, warnings, errors) {
       let pattern = /^([a-zA-Z][a-zA-Z0-9]*)?$/;
       if (!pattern.test(name)) {
-        errors.push('The application name must begin with a letter and must contain only letters or digits. No ' +
-          'special characters are allowed.');
+        pattern = /^([a-zA-Z][a-zA-Z0-9-]*)?$/;
+        if (pattern.test(name)) {
+          warnings.push('Dashes should only be used in application names when using the Kubernetes v2 provider.');
+        } else {
+          errors.push('The application name must begin with a letter and must contain only letters or digits. ' +
```

Comment:
This error message is incorrect with this change.  Should it mention that dashes are allowable?

Reply:
I decided against it since the error will only be shown if the name includes something which is not a dash. For example, "foo!" shows the error, "foo-" (even while typing) shows the warning, "foobar" shows no message, "foo-bar" just shows the warning.

By the time this error is shown, I feel like the error is sufficiently clear to the user, and adding an "except dashes" to the error (which will then cause a warning about dashes to be shown) doesn't seem helpful.